### PR TITLE
Issue/#1662 helm improvement

### PR DESCRIPTION
**Pull request by thyarles**
I'm working on a fix to https://github.com/gravitee-io/issues/issues/1662, until it's done I made some helm improvements like security, possibility to config LDAP in values.yaml and fix of other errors that I found when was testing the Helm version. 

I tested this version in two Kubernetes clusters, one on cloud (GKE) and other on-premise with success. 

Now it's easy to configure your LDAP (with detailed instructions) and to avoid the creation of users that are not configured in your values.xml.

### CODE REVIEW

File: apim/templates/api-configmap.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ data:
           authentication:
             type: basic
             users:
-              admin: adminadmin
+              admin: admin
```
Comment (brasseld) [inline apim/templates/api-configmap.yaml]: Why updating the password here?

File: apim/values.yaml [hunk 2]
```diff
@@ -7,21 +7,73 @@ chaos:
...
-#     password: $2a$10$2gtKPYRB9zaVaPcn5RBx/.3T.7SeZoDGs9GKqbo9G64fKyXFR1He.
-
 # Default password "admin", use bcrypt ($2a$ version) to generate a new one
 adminPasswordBcrypt: $2a$10$Ihk05VSds5rUSgMdsMVi9OKMIx2yUvMz7y9VP3rJmQeizZLrhLMyq
 
 jwtSecret: myJWT4Gr4v1t33_S3cr3t
 
+# Define extra inMemory users here or disable the default ones here
+extraInMemoryUsers: |
+  - user:
```
Comment (brasseld) [inline apim/values.yaml]: Please keep the existing user examples:

```
#   - user:	
#     username: publisher-user	
#     password: $2a$13$iXdXO4wAYdhx2LOwijsp7.PsoAZQ12zEdHxbriIYCbtyo.y32LTji	
#     roles: MANAGEMENT:API_PUBLISHER, PORTAL:API_PUBLISHER	
#   - user:	
#     username: application-test	
#     password: $2a$10$2gtKPYRB9zaVaPcn5RBx/.3T.7SeZoDGs9GKqbo9G64fKyXFR1He.
```

File: apim/values.yaml [hunk 3]
```diff
@@ -7,21 +7,73 @@ chaos:
...
+    password: $2a$10$iXdXO4wAYdhx2LOwijsp7.PsoAZQ05zEdHxbriIYCbtyo.y32LTji
+    # You can declare multiple roles using comma separator
+    roles: MANAGEMENT:API_PUBLISHER, PORTAL:API_PUBLISHER
+  - user:
+    username: application1
+    # Password value: application1
+    password: $2a$10$2gtKPYRB9zaVaPcn5RBx/.3T.7SeZoDGs9GKqbo9G64fKyXFR1He.
+    roles: MANAGEMENT:USER, PORTAL:USER
+
+inLdapAuth:
```
Comment (brasseld) [inline apim/values.yaml]: I suppose you can rename this key to something like `ldap`

File: apim/values.yaml [hunk 4]
```diff
@@ -7,21 +7,73 @@ chaos:
...
+  enabled: false
+  context:
+    # User to bind the LDAP
+    bindUser: user@example.com
```
Comment (brasseld) [inline apim/values.yaml]: `bindUser` -> `user`

File: apim/values.yaml [hunk 5]
```diff
@@ -7,21 +7,73 @@ chaos:
...
+    # Password to bind the LDAP
+    bindPassword: pass@12345
```
Comment (brasseld) [inline apim/values.yaml]: `bindPassword` -> `password`

File: apim/values.yaml [hunk 6]
```diff
@@ -7,21 +7,73 @@ chaos:
...
+    # URL to LDAP
+    bindUrl: ldap://ldap.example.com
```
Comment (brasseld) [inline apim/values.yaml]: `bindUrl` -> `url`

File: apim/values.yaml [hunk 7]
```diff
@@ -7,21 +7,73 @@ chaos:
...
+    # Bind base to be used in authentication and lookup sections
+    bindBase: dc=example,dc=com
```
Comment (brasseld) [inline apim/values.yaml]: `bindBase` -> `base`

File: apim/templates/api-configmap.yaml (hunk 1 above)
Comment (thyarles) [reply apim/templates/api-configmap.yaml]: My bad, I changed the password to test and commit by mistake. I'll revert it.

File: apim/values.yaml (hunk 2 above)
Comment (thyarles) [reply apim/values.yaml]: Sure!

File: apim/values.yaml (hunk 6 above)
Comment (thyarles) [reply apim/values.yaml]: Done!

**Comment by brasseld** [comment]
Perfect, can you please squash your 8 commits to a single one?

**Comment by thyarles** [comment]
Hello @brasseld,

Thanks for taking time to review the code and make suggestions.

Please, let me know if I have to do anything else.

**Comment by brasseld** [comment]
LGTM

I should run more tests before validate this pull request

**Comment by brasseld** [comment]
Thanks a lot @thyarles!

**Pull request merged**